superset-frontend/src/views/CRUD/data/database/DatabaseModal/DatabaseConnectionForm.tsx

```diff
@@ -437,30 +437,45 @@ const forceSSLField = ({
   changeMethods,
   db,
   sslForced,
-}: FieldPropTypes) => (
-  <div css={(theme: SupersetTheme) => infoTooltip(theme)}>
-    <Switch
-      disabled={sslForced && !isEditMode}
-      checked={db?.parameters?.encryption || sslForced}
-      onChange={changed => {
-        changeMethods.onParametersChange({
-          target: {
-            type: 'toggle',
-            name: 'encryption',
-            checked: true,
-            value: changed,
-          },
-        });
-      }}
-    />
-    <span css={toggleStyle}>SSL</span>
-    <InfoTooltip
-      tooltip={t('SSL Mode "require" will be used.')}
-      placement="right"
-      viewBox="0 -5 24 24"
-    />
-  </div>
-);
+}: FieldPropTypes) => {
+    return (
+      <div css={(theme: SupersetTheme) => infoTooltip(theme)}>
+        <Switch
+          disabled={sslForced && !isEditMode}
+          checked={db?.parameters?.encryption || sslForced}
+          onChange={changed => {
+            changeMethods.onParametersChange({
+              target: {
+                type: 'toggle',
+                name: 'encryption',
+                checked: true,
+                value: changed,
+              },
+            });
+          } } />
+        <span css={toggleStyle}>SSL</span>
+
+        {db?.database_name === 'PostgreSQL' && (
+          <InfoTooltip
+            tooltip={t('Requires a root certificate authority (public, local, or self-signed). SSL Mode "require" will be used.')}
+            placement="right"
+            viewBox="0 -5 24 24" />)}
+
+        {db?.database_name === 'MySQL' && (
+          <InfoTooltip
+            tooltip={t('Requires a root certificate authority (public, local, or self-signed).')}
+            placement="right"
+            viewBox="0 -5 24 24" />)}
+
+        {db?.database_name === 'Amazon Redshift' && (
+          <InfoTooltip
+            tooltip={t('Requires a root certificate authority (public, local, or self-signed). SSL Mode "verify-ca" will be used.')}
+            placement="right"
+            viewBox="0 -5 24 24" />)}
+
```

Comment:
Thanks for this @gabester78! One suggestion that could make this database type agnostic and also clean up the three different InfoTooltips here would be to 
1) use an object for getting the different text, so something like: 

```
const tooltipText = {
    foo: "some text",
    bar: "other text",
}
```

and then 
```
<InfoTooltip
            tooltip={t(tooltipText(somevariable))}
            placement="right"
            viewBox="0 -5 24 24" />)}
```

Reply:
2. you should be able to use the db.encryption_params.sslmode to switch between the different types of text, so you would pass the key in like `tooltip={t(tooltipText(db.encryption_params.sslmode))}`

Reply:
Great thanks! Should I go ahead and make the changes?

Reply:
Closing PR. Reference [](https://github.com/apache/superset/pull/16907) for updates.